tasks/api/src/main/java/org/projectnessie/nessie/tasks/api/TaskState.java

```diff
@@ -98,19 +124,22 @@ default void check() {
       case SUCCESS:
         checkState(retryNotBefore() == null, "retryNotBefore must be null for SUCCESS");
         checkState(lostNotBefore() == null, "retryNotBefore must be null for SUCCESS");
+        checkState(deadline() == null, "deadline must be null for SUCCESS");
         break;
       case FAILURE:
-        checkState(retryNotBefore() == null, "retryNotBefore must be null for FAILURE");
```

Comment:
Maybe just `!=` instead of removing?

Reply:
Old tasks loaded from storage can have `null` values here.

Reply:
removed this from current PR... will follow-up separately.